### ISSUE TITLE
Wrong Hook Context for Nested Operations

### ISSUE THREAD

**Issue by frank-trampe** [feathersjs/feathers#1630]
In version 2.2.4 (the latest minor release from major version 2), an error-handling hook receives the context of the operation generating the error instead of the context for its position in the call stack.

So if
* operations A and B both occur in services with error hooks,
* A depends on B,
* and B throws an error,
then the error handlers for operations A and B both receive the hook context for B.

To test, just set an extra value in hook.params in each service and log the extra value from the error handler for A.

Is there any chance of a maintenance update to fix this? I expect and hope that it would be a one-line change.

**Comment by frank-trampe** [comment]
I figured out how this is happening. On line 101 [here](https://github.com/feathersjs-ecosystem/feathers-hooks/blob/v2.1.2/src/hooks.js), the hook context is set from the "hook" property of the error to be handled. So, even if the error is bubbling up, the upper levels still get the hook context of the error instead of their own contexts. What would be wrong with using just `hookObject`?

**Comment by daffl** [comment]
This has been fixed in later versions. Your two options are

1. Upgrade to a later version. A lot of work went into ensuring a smooth migration path between versions.
2. Contact me on options for a support contract that includes support and releases for legacy version

**Comment by frank-trampe** [comment]
Ah, okay. It sounds from that like the underlying issue is not as simply fixed as I proposed. I've just worked around it by adding an error hook that removes the hook context from the error for each service, which seems to work for my use cases.

```
function hookErrorStripHookContext(hook) {
	if (hook.error instanceof Error && hook.error.hook instanceof Object) {
		delete hook.error.hook;
	}
	return hook;
}
```

I do hope to upgrade at some point, of course, but I run into the edge cases whatever I am doing, and the bumps from migrating seem likely to be bigger than from patching the few little issues like this.
